File: app/src/main/java/com/numero/sojodia/model/Time.java
package com.numero.sojodia.model;

import android.annotation.SuppressLint;

import java.util.Calendar;

public class Time {

    private int hour;
    private int min;
    private int sec;

    public Time() {
        Calendar calendar = Calendar.getInstance();
        this.hour = calendar.get(Calendar.HOUR_OF_DAY);
        this.min = calendar.get(Calendar.MINUTE);
        this.sec = calendar.get(Calendar.SECOND);
    }

    public Time(int hour, int min, int sec) {
        this.hour = hour;
        this.min = min;
        this.sec = sec;
    }

    public void setTime(int hour, int min, int sec) {
        this.hour = hour;
        this.min = min;
        this.sec = sec;
    }

    public int getHour() {
        return hour;
    }

    public int getMin() {
        return min;
    }

    public int getSec() {
        return sec;
    }

    public String toString() {
        return toString(true);
    }

    @SuppressLint("DefaultLocale")
    public String toString(boolean isShowSec) {
        if (isShowSec) {
            return String.format("%02d:%02d:%02d", hour, min, sec);
        }
        return String.format("%02d:%02d", hour, min);
    }
}

File: model/src/main/java/com/numero/sojodia/model/Time.kt
package com.numero.sojodia.model

import java.util.*

class Time {

    val hour: Int
    val min: Int

    constructor() {
        val calendar = Calendar.getInstance()
        hour = calendar.get(Calendar.HOUR_OF_DAY)
        min = calendar.get(Calendar.MINUTE)
    }

    constructor(hour: Int, min: Int) {
        this.hour = hour
        this.min = min
    }

    operator fun compareTo(time: Time): Int {
        if (hour == time.hour && min == time.min) return 0
        if ((hour == time.hour && min <= time.min) or
                (hour < time.hour)) return -1
        return 1
    }

    operator fun minus(time: Time): Time {
        var hour = this.hour - time.hour
        var min = this.min - time.min

        if (min < 0) {
            min += 60
            hour -= 1
        }

        if (hour < 0) {
            hour += 24
        }
        return Time(hour, min)
    }
}
